### PR TITLE
fuzz: Add check in `p2p_headers_presync` that chain work never exceeds minimum work

### PR DESCRIPTION
**Pull request by marcofleon**
A followup to https://github.com/bitcoin/bitcoin/pull/30661

The added assertion just makes sure that the fuzz test is working as intended. If we're sure that the total work of the test chain is never more than minimum chain work, then we can be sure that the later assertion failure would actually mean that a bug in the headers presync logic was found.

This PR also addresses:
https://github.com/bitcoin/bitcoin/pull/30661#discussion_r1746614616
https://github.com/bitcoin/bitcoin/pull/30661#discussion_r1764943665
https://github.com/bitcoin/bitcoin/pull/30661#discussion_r1764961991

### CODE REVIEW

**Review by instagibbs — APPROVED** [review]
ACK 7b85fd1443488afe60235e4c33bd956cb9c8d562

Thanks for the quick follow-up on suggestion! Makes the test significantly easier to read.

File: src/test/fuzz/p2p_headers_presync.cpp [hunk 1]
```diff
@@ -179,16 +185,25 @@ FUZZ_TARGET(p2p_headers_presync, .init = initialize)
...
+                all_headers.push_back(block);
+
                 auto headers_msg = NetMsg::Make(NetMsgType::BLOCK, TX_WITH_WITNESS(block));
                 g_testing_setup->SendMessage(fuzzed_data_provider, std::move(headers_msg));
             });
+
+        total_work += CalculateClaimedHeadersWork(all_headers);
+
+        // This test should never create a chain with more work than MinimumChainWork.
+        assert(total_work < chainman.MinimumChainWork());
```
[the remark this answers is not in the document]
Comment (marcofleon) [reply src/test/fuzz/p2p_headers_presync.cpp:207]: That's true, good note.

**Comment by marcofleon** [comment]
I also updated the PR description to include some additional changes.

**Comment by instagibbs** [comment]
reACK 284bd17309ab3b124d9dcddfec62f5506383343b

**Comment by achow101** [comment]
ACK 284bd17309ab3b124d9dcddfec62f5506383343b

**Pull request merged**